tests/test_json.py

```diff
@@ -376,3 +376,22 @@ def test_partial_parse():
     with pytest.raises(ValueError, match='EOF while parsing a string at line 1 column 15'):
         from_json(b'["aa", "bb", "c')
     assert from_json(b'["aa", "bb", "c', allow_partial=True) == ['aa', 'bb']
+
+
+def test_json_bytes_base64_round_trip():
+    data = b'hello'
+    encoded = b'"aGVsbG8="'
+    assert to_json(data, bytes_mode='base64') == encoded
+
+    v = SchemaValidator({'type': 'bytes'}, {'val_json_bytes': 'base64'})
+    assert v.validate_json(encoded) == data
+
+    with pytest.raises(ValueError):
+        v.validate_json('"wrong!"')
```

Comment:
I think we probably need to expect a `ValidationError` here, where the input was bytes but the wrong format. It might mean adding a new error type e.g. `bytes_format`.